src/mmap_unix.rs

```diff
@@ -289,23 +288,6 @@ impl AsSlice for MmapRegion {
     }
 }
 
-impl VolatileMemory for MmapRegion {
```

Comment:
This and related changes to simplify current implementation deserves a dedicated PR so we could do it step by step:)